config/config.properties

```diff
@@ -20,162 +20,162 @@
-#pravegaservice.storageThreadPoolSize=200
+#pravegaservice.threadPool.storage.size=200
 
 # TCP port where the SegmentStore will be listening for incoming requests.
 # Valid values: Positive integer in the valid TCP port ranges.
-pravegaservice.listeningPort=12345
+pravegaservice.service.listener.port=12345
 
 # Full URL (host:port) where to find a ZooKeeper that can be used for coordinating this Pravega Cluster.
 # Required.
-pravegaservice.zkURL=localhost:2181
+pravegaservice.zk.connect.uri=localhost:2181
 
 # Number of milliseconds to wait between ZooKeeper retries.
 # Valid values: Positive integer.
 # Recommended values: Between 5000 and 10000 (too high will cause failover delays).
-#pravegaservice.zkRetrySleepMs=5000
+#pravegaservice.zk.connect.retries.interval.milliseconds=5000
 
 # Number of times to retry a failed attempt to contact ZooKeeper.
 # Valid values: Positive integer.
 # Recommended values: Between 3 and 5.
-#pravegaservice.zkRetryCount=5
+#pravegaservice.zk.connect.retries.count.max=5
 
 # Session timeout for ZooKeeper.
 # Valid values: Positive integer.
-#pravegaservice.zkSessionTimeoutMs=10000
+#pravegaservice.zk.connect.sessionTimeout.milliseconds=10000
 
 # Enable security for the ZK client.
-#pravegaservice.secureZK=false
+#pravegaservice.zk.connect.security.enable=false
 
 # Location of trust store for secure connection to ZK.
-#pravegaservice.zkTrustStore=
+#pravegaservice.zk.connect.security.tls.trustStore.location=
 
 #Location of password file to access the trust store.
-#pravegaservice.zkTrustStorePasswordPath=
+#pravegaservice.zk.connect.security.tls.trustStore.pwd.location=
 
-# DataLog implementation for Tier 1 storage.
+# DataLog implementation for Durable Data Log Storage.
 # Valid values: BOOKKEEPER, INMEMORY.
 # Default value: BOOKKEEPER
-pravegaservice.dataLogImplementation=BOOKKEEPER
+pravegaservice.dataLog.impl.name=BOOKKEEPER
 
-# Storage implementation for Tier 2 storage.
+# Storage implementation for Long-Term Storage.
 # Valid values: HDFS, FILESYSTEM, EXTENDEDS3, INMEMORY.
 # Default value: HDFS
-# pravegaservice.storageImplementation=HDFS
+# pravegaservice.storage.impl.name=HDFS
 
 # Storage NO-OP Mode: in No-Op mode, user stream segment writing is no-oped; user stream segment reading is not supported.
 # This mode is used to avoid storage interference in testing while still keep the system functioning as usual.
-# NOTE: pravegaservice.storageImplementation is still used to store metadata and system segments, which are required for the functioning of the Pravega Cluster.
+# NOTE: pravegaservice.storage.impl.name is still used to store metadata and system segments, which are required for the functioning of the Pravega Cluster.
 # WARNING: Do NOT set NO-OP Mode to true in production environment! Data loss will happen!
 # Valid values: true, false
 # Default value: false
-#storageextra.storageNoOpMode=false
+#storageextra.noOp.mode.enable=false
```

Comment:
Since this was introduced, I think that this could me more understandable as `pravegaservice.storage.noOp.mode.enable`.

Reply:
This config property is defined in `StorageExtraConfig.java` that has a component code `storageextra`, which is resulting in the `storageextra` prefix. Refactoring it to bring it under `pravegservice` (`ServiceConfig.java`) would require kind of changes that are out-of-scope of this PR. We only intend to change configuration property keys/names here, and that too in a backward compatible manner.